src/Util/GeometryUtil/GeometryUtil.js

```diff
@@ -0,0 +1,292 @@
+   *  Default is to EPSG:3857.
+   * @param {Number} tolerance The tolerance (in meters) used to find if a line
+   *  vertex is inside the polygon geometry. Default is to 10.
+   *
+   * @returns {ol.Feature[] | ol.geom.Polygon[]} An array of instances of ol.feature
+   *  with/or ol.geom.Polygon
+   */
+  static splitByLine(polygon, line, projection = 'EPSG:3857', tolerance = 10) {
+    const geoJsonFormat = new OlFormatGeoJSON({
+      dataProjection: 'EPSG:4326',
+      featureProjection: projection
+    });
+    const polygonFeat = polygon instanceof OlFeature ? polygon
+      : new OlFeature({
+        geometry: polygon
+      });
+    const lineFeat = line instanceof OlFeature ? line
+      : new OlFeature({
+        geometry: line
+      });
+
+    // Convert the input features to turf.js/GeoJSON features while
+    // reprojecting them to the internal turf.js projection 'EPSG:4326'.
+    const turfPolygon = geoJsonFormat.writeFeatureObject(polygonFeat);
+    const turfLine = geoJsonFormat.writeFeatureObject(lineFeat);
+
+    // Union both geometries to a set of MultiLineStrings.
+    const unionGeom = union(polygonToLine(turfPolygon), turfLine);
+
+    // Buffer the input polygon to take great circle (un-)precision
+    // into account.
+    const bufferedTurfPolygon = buffer(turfPolygon, tolerance, {
+      units: 'meters'
+    });
+
+    let filteredSegments = [];
+
+    // Iterate over each segment and remove any segment that is not covered by
+    // the (buffered) input polygon.
+    segmentEach(unionGeom, (currentSegment, featureIndex, multiFeatureIndex) => {
+      const segmentCenter = centroid(currentSegment);
+      const isSegmentInPolygon = booleanPointInPolygon(segmentCenter, bufferedTurfPolygon);
+
+      if (isSegmentInPolygon) {
+        if (!filteredSegments[multiFeatureIndex]) {
+          filteredSegments[multiFeatureIndex] = [];
+        }
+
+        if (filteredSegments[multiFeatureIndex].length === 0) {
+          filteredSegments[multiFeatureIndex].push(
+            getCoords(currentSegment)[0],
+            getCoords(currentSegment)[1]
+          );
+        } else {
+          filteredSegments[multiFeatureIndex].push(
+            getCoords(currentSegment)[1]
+          );
+        }
+      }
+    });
+
+    // Rebuild the unioned geometry based in the filtered segments.
+    const filteredUnionGeom = multiLineString(filteredSegments);
+
+    // Polygonize the lines.
+    const polygonizedUnionGeom = polygonize(filteredUnionGeom);
+
+    // Return as Array of ol.Features.
+    const features = geoJsonFormat.readFeatures(polygonizedUnionGeom);
+    if (polygon instanceof OlFeature && line instanceof OlFeature) {
+      return features;
+    } else {
+      return features.map(f => f.getGeometry());
+    }
+  }
+
+  /**
+   * Adds a buffer to a given geometry.
+   *
+   * @param {ol.geom.Geometry | ol.Feature} geometry The geometry.
+   * @param {Integer} buffer The buffer to add in meters.
```

Comment:
Integer is not a valid type I guess. what happens if a float like number is passed? can we use this to dissolve a feature with a negative value? will buffering consider holes in polygons?